service/src/main/java/org/mskcc/smile/service/impl/TempoMessageHandlingServiceImpl.java

```diff
@@ -364,63 +367,79 @@ public void run() {
+
                 // confirm tempo data exists by primary id
-                Tempo tempo = tempoService.getTempoDataBySamplePrimaryId(primaryId);
-                if (tempo == null) {
+                if (tempoSample == null) {
                     LOG.error("Tempo data not found for sample with Primary ID " + primaryId);
                     continue;
                 }
-                // validate props before building tempo sample
-                SampleMetadata sampleMetadata = sampleService.getLatestSampleMetadataByPrimaryId(primaryId);
-                String cmoSampleName = sampleMetadata.getCmoSampleName();
-                if (StringUtils.isBlank(cmoSampleName)) {
+                // validate props before adding tempo sample to set of samples to be published
+                if (StringUtils.isBlank(tempoSample.getCmoSampleName())) {
                     LOG.error("Invalid CMO Sample Name for sample with Primary ID " + primaryId);
                     continue;
                 }
-                String accessLevel = tempo.getAccessLevel();
-                if (StringUtils.isBlank(accessLevel)) {
+                if (StringUtils.isBlank(tempoSample.getAccessLevel())) {
                     LOG.error("Invalid Access Level for sample with Primary ID " + primaryId);
                     continue;
                 }
-                String custodianInformation = tempo.getCustodianInformation();
-                if (StringUtils.isBlank(custodianInformation)) {
+                if (StringUtils.isBlank(tempoSample.getCustodianInformation())) {
                     LOG.error("Invalid Custodian Information for sample with Primary ID " + primaryId);
                     continue;
                 }
-                String cmoPatientId = sampleMetadata.getCmoPatientId();
-                SmilePatient patient = patientService.getPatientByCmoPatientId(cmoPatientId);
-                // build tempo sample object
-                TempoSample tempoSample = TempoSample.newBuilder()
-                    .setPrimaryId(primaryId)
-                    .setCmoSampleName(cmoSampleName)
-                    .setAccessLevel(accessLevel)
-                    .setCustodianInformation(custodianInformation)
-                    .setBaitSet(StringUtils.defaultString(sampleMetadata.getBaitSet(), ""))
-                    .setGenePanel(StringUtils.defaultString(sampleMetadata.getGenePanel(), ""))
-                    .setOncotreeCode(StringUtils.defaultString(sampleMetadata.getOncotreeCode(), ""))
-                    .setCmoPatientId(StringUtils.defaultString(cmoPatientId, ""))
-                    .setDmpPatientId(StringUtils.defaultString(patient.getPatientAlias("dmpId"), ""))
-                    .setRecapture(sampleService.sampleIsRecapture(sampleMetadata.getInvestigatorSampleId()))
-                    .build();
                 validTempoSamples.add(tempoSample);
```

Comment:
I've consolidated the multiple calls we were making to the patient, tempo, and sample services into one call within the tempo service.

The new query here: https://github.com/mskcc/smile-server/pull/1459/files#diff-049c82a9f9eac7032dbfbd28b5ebdfcfe215367499ecb7d485d5324b8b82578dR133

creates a `Map` of the result and then the call to `tempoService.getTempoSampleDataBySamplePrimaryId(..)` handles the actual building of the `TempoSample` object 

@qu8n